core/src/main/scala/io/projectglow/transformers/blockvariantsandsamples/VariantSampleBlockMaker.scala

```diff
@@ -20,13 +20,28 @@ import io.projectglow.common.GlowLogging
 import io.projectglow.common.VariantSchemas._
 import io.projectglow.functions._
 
-import org.apache.spark.sql.DataFrame
+import org.apache.commons.lang3.reflect.FieldUtils
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{AssertTrue, BinaryExpression, Expression}
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
 import org.apache.spark.sql.expressions.Window
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.types.{ArrayType, IntegerType, StringType}
 
 private[projectglow] object VariantSampleBlockMaker extends GlowLogging {
 
+  def validateNumValues(df: DataFrame): DataFrame = {
+    import df.sparkSession.implicits._
+    val expectedNumValues =
+      df.selectExpr("size(values) as numValues").take(1)(0).getAs[Int]("numValues")
+    df.filter(
+      isnull(
+        assert_true_or_error(
+          size(col("values")) === expectedNumValues,
+          "Number of values is inconsistent!")))
```

Comment:
How about "At least one row has an inconsistent number of values (expected x). Please verify that each row contains the same number of values."